src/common/tracer.cc

```diff
@@ -55,6 +65,37 @@ bool Tracer::is_enabled() const {
   return g_ceph_context->_conf->jaeger_tracing_enable;
 }
 
+void encode(const jspan_context& span_ctx, bufferlist& bl, uint64_t f) {
+  if (span_ctx.IsValid()) {
+    using namespace opentelemetry;
+    using namespace trace;
+    ENCODE_START(1, 1, bl);
+    encode_nohead(std::string_view((char*)span_ctx.trace_id().Id().data(), TraceId::kSize), bl);
```

Comment:
what is the type of `span_ctx.trace_id().Id().data()` ?
we should avoid c-style casts.
the problem with these casts is that if they change something in the library later, causing a problem with the cast, we would probably not get any compilation error.